src/psyclone/core/signature.py

```diff
@@ -116,21 +116,33 @@ def __ne__(self, other):
     # ------------------------------------------------------------------------
     def __lt__(self, other):
         '''Required to sort signatures. It just compares the tuples.'''
+        if not isinstance(other, Signature):
+            raise TypeError("'<' not supported between instances of "
+                            "'Signature' and '{0}'.".format(type(other)))
```

Comment:
Sorry to be picky (I know) but if you do `type(other).__name__` then the result is independent of python 2/3.

Reply:
Good point, done. And I know you mentioned not bothering with the test for the texts, but I've fixed them anyway since only then I felt I could remove the 'python2 vs python3' comment.